src/main/scala/firrtl/passes/AnnotateMemMacros.scala

```diff
@@ -0,0 +1,167 @@
+package firrtl.passes
+
+import scala.collection.mutable.{ArrayBuffer,HashMap}
+import AnalysisUtils._
+import firrtl.WrappedExpression._
+import firrtl.ir._
+import firrtl._
+import firrtl.Utils._
+import firrtl.Mappers._
+
+case class AppendableInfo(fields: Map[String,Any]) extends Info {
+  def append(a: Map[String,Any]) = this.copy(fields = fields ++ a)
+  def append(a: (String,Any)): AppendableInfo = append(Map(a))
+  def get(f: String) = fields.get(f)
+  override def equals(b: Any) = b match {
+    case i: AppendableInfo => fields - "info" == i.fields - "info"
+    case _ => false
+  }
+}
+
+object AnalysisUtils {
+
+  def getConnects(m: Module) = {
+    val connects = HashMap[String, Expression]()
+    def getConnects(s: Statement): Unit = s match {
+      case s: Connect =>
+        connects(s.loc.serialize) = s.expr
+      case s: DefNode =>
+        connects(s.name) = s.value
+      case s: Block =>
+        s.stmts foreach getConnects
+      case _ =>
+    }
```

Comment:
So this code is not going to be finding connects in Conditionals, if that's intentional, no worries, if it's not intentional though we should fix it. It general how I would write this is (still using connects from above and unapply matching which I generally prefer but is just a stylistic opinion):
```scala
    def getConnects(s: Statement): Statement = {
      s map getConnects match {
        case Connect(_, loc, expr) =>
          connects(loc.serialize) = expr
        case DefNode(_, name, value) =>
          connects(name) = value
        case _ => // do nothing
      }
      s // return because we only have map and not foreach right now
    }
```